core/app/models/spree/order.rb

```diff
@@ -424,21 +424,11 @@ def ensure_line_items_are_in_stock
     end
 
     def empty!
-      if completed?
```

Comment:
@KacperMekarski also we need to deprecate this method

```
ActiveSupport::Deprecation.warn(<<-DEPRECATION, caller)
  `Order#empty!` is deprecated and will be removed in Spree 5.0.
  Please use `Spree::Cart::Empty.call(order: order)` instead.
DEPRECATION
```

we also need to migrate to this new behaviour everywhere in the codebase

Reply:
done.